frontend/src/utils/timerange.js

```diff
@@ -0,0 +1,197 @@
+import { DateTime } from "luxon";
+
+const NANOS_PER_SECOND = 1_000_000_000n;
+const TIMERANGE_REGEX =
+  /^(?<startInclusive>\[|\()?(?:(?<startSeconds>-?(?:0|[1-9][0-9]*)):(?<startNanos>0|[1-9][0-9]{0,8}))?(?:_(?:(?<endSeconds>-?(?:0|[1-9][0-9]*)):(?<endNanos>0|[1-9][0-9]{0,8}))?)?(?<endInclusive>\]|\))?$/;
+const INFINITE_PATTERNS = new Set([
+  "_",
+  "(_)",
+  "[_]",
+  "(_]",
+  "[_)",
+  "(_",
+  "[_",
+  "_)",
+  "_]",
+]);
+const EMPTY_PATTERNS = new Set([
+  "",
+  "()",
+  "[]",
+  "(]",
+  "[)",
+  ")",
+  "]",
+  "(",
+  "[",
+]);
+
+const isInclusive = (bracket) => bracket !== "(";
+const isEndInclusive = (bracket) => bracket !== ")";
+
+const createTimerangeResult = (includesStart, start, end, includesEnd) => ({
```

Comment:
is it worth adding some defaults here and then you don't always need to provide all arguments?

Reply:
I did think about this, however i decided against it as the original code that had all the bugs in it had some bugs because we had defaults and it was not clear they were defaulting hence the bugs. So i decided in this case to make it explicit each time